nntrainer/layers/preprocess_l2norm_layer.cpp

```diff
@@ -39,14 +39,15 @@ void PreprocessL2NormLayer::finalize(InitLayerContext &context) {
 
 void PreprocessL2NormLayer::forwarding(RunLayerContext &context,
                                        bool training) {
+  const float epsilon = std::get<props::Epsilon>(l2norm_props);
   auto &hidden_ = context.getOutput(SINGLE_INOUT_IDX);
 
   auto &input_ = context.getInput(SINGLE_INOUT_IDX);
 
   for (unsigned int b = 0; b < input_.batch(); ++b) {
     auto input_slice = input_.getBatchSlice(b, 1);
     auto hidden_slice = hidden_.getBatchSlice(b, 1);
-    input_slice.multiply(1 / input_slice.l2norm(), hidden_slice);
+    input_slice.multiply(1 / (input_slice.l2norm() + epsilon), hidden_slice);
```

Comment:
Wouldn't this impact cases where the L2 norm is not zero?

Reply:
Good point. Let me check it in terms of our layer's definition.
I thought the preprocess_l2norm_layer is related to LayerNorm of pytorch [[link](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html)]. However, if it is not the layer norm but normalization function, I need to update it like
```suggestion
    input_slice.multiply(1 / max(input_slice.l2norm(), epsilon), hidden_slice);
```
Please let me check it and make another PR if it corresponds to the latter case.